src/system_data.jl

```diff
@@ -851,6 +866,11 @@ end
 get_components_by_name(::Type{T}, data::SystemData, args...) where {T} =
     get_components_by_name(T, data.components, args...)
 
+function get_components(data::SystemData, attribute::SupplementalAttribute)
```

Comment:
Are we ok with letting the compiler figure out the `eltype` of the resulting vector or should we add a type annotation?

Reply:
The eltype could be concrete or abstract depending on whether all components are the same type. The compiler should make the eltype the concrete type or first common supertype.